tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/Mocks/IDebugLaunchProviderFactory.cs

```diff
@@ -1,20 +1,26 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
 
 using System;
+using Microsoft.VisualStudio.ProjectSystem.VS.Debug;
 using Moq;
 
 namespace Microsoft.VisualStudio.ProjectSystem.Debug
 {
     public static class IDebugLaunchProviderFactory
     {
-        public static IDebugLaunchProvider ImplementCanLaunchAsync(Func<bool> action)
+        public static IDebugLaunchProvider ImplementIsProjectDebuggableAsync(Func<bool> action)
         {
-            var mock = new Mock<IDebugLaunchProvider>();
+            var mock = new Mock<IDebugLaunchProviderMock>();
 
-            mock.Setup(d => d.CanLaunchAsync(It.IsAny<DebugLaunchOptions>()))
+            mock.Setup(d => d.CanBeStartupProjectAsync(It.IsAny<DebugLaunchOptions>()))
                                 .ReturnsAsync(action);
 
             return mock.Object;
         }
     }
+
+    internal interface IDebugLaunchProviderMock : IDebugLaunchProvider, IStartupProjectProvider
+    {
+
+    }
```

Comment:
Moq can create mocks that implement multiple interfaces, so this is not needed here:

```c#
// implementing multiple interfaces in mock
var mock = new Mock<IFoo>();
var disposableFoo = mock.As<IDisposable>();
// now the IFoo mock also implements IDisposable :)
disposableFoo.Setup(disposable => disposable.Dispose());
```

From https://github.com/Moq/moq4/wiki/Quickstart#advanced-features